ispell improvements

rrthomas opened the pull request:
Add support for most ispell prefix commands, allowing programs such as Emacs
to use enchant as a full ispell replacement, adding words to the session
word list and personal word list.

rrthomas committed:
enchant-ispell.c: Fix spelling of “tuple” in comments

rrthomas committed:
enchant-ispell.c: Add support for most ispell prefix commands

This allows programs such as Emacs to use enchant as a full ispell
replacement, adding words to the session word list and personal word list.

rrthomas committed:
enchant-ispell: implement $$ra (aspell extension)

This command allows a replacement to be set for the rest of a session.

hfiguiere commented:
There is now Travis CI setup for enchant. If you add commits to the branch it will run automatically.

hfiguiere commented on tests/enchant-ispell.c:
```diff
@@ -421,31 +427,94 @@ parse_file (FILE * in, FILE * out, IspellMode_t mode, int countLines, gchar *dic
-				pos = GPOINTER_TO_INT(tokens->data);
-				tokens = tokens->next;
+			if (mode == MODE_A) {
+				switch ((int)*str->str) {
```
is there a reason to do a cast here since `str->str` is a `gchar *`?

rrthomas replied:
Good spot, that's a redundant cast left over from an earlier version of the code. Will fix. Thanks for setting up Travis.

rrthomas committed:
Remove redundant cast

rrthomas committed:
enchant-ispell: minor whitespace fix

rrthomas committed:
enchant-ispell: allow apostrophes anywhere

This applies to informal English (e.g. ‘asn’t, h-less hasn’t), and of course
some other languages, e.g. Dutch.

Add a note about deprecation of G_UNICODE_COMBINING_MARK.

Reorganise is_word_char to put code in more logical order: test for
apostrophe at the top (include curly apostrophe), as its GUnicodeType is not
needed.

rrthomas commented:
I just noticed that enchant-lsmod already lists dictionaries, so I removed my change to make enchant-ispell do that.

I have added a patch to enchant-ispell to allow apostrophes anywhere, as this is required in at least English and Dutch. I also added support for the Unicode curly apostrophe.

rrthomas commented:
I just fixed a bug with the handling of `^` at the start of a line, where the wrong position was reported for mis-spellings (out by 1 error).
